infrastructure/ansible/dynamic.inventory/TO.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Generate an ansible inventory compatable dataset from TO
+
+Before using, make sure you have the TrafficOps Python Native Client library available inside your env
+# Usage: python TO.py -to <TO_USER> <TO_PASS> -url to.kabletown.invalid --list
+export TO_USERNAME=<TO_USERname>
+export TO_PASSWORD=<TO_PASSword>
+export TO_URL=<TO_URL>
+"""
+
+import json
+import argparse
+import logging
+import os
+import collections
+from trafficops.tosession import TOSession
+
+# Disable rest api logging, allowing invalid certs causes warnings
+logging.getLogger('common.restapi').disabled = True
+#logging.getLogger('common.restapi').logging_level = logging.DEBUG
+
+def empty_inventory():
+    """Generate a valid empty inventory"""
+    return {'_meta': {'hostvars': {}}}
+
+
+class AnsibleInventory(object):
+    """Wrapper class for needed methods"""
+
+    def __init__(self, user, password, url):
+        """Init base members"""
+        self.to_user = user
+        self.to_pass = password
+        self.to_url = url
+
+    @classmethod
+    def populate_server_profile_vars(cls, api, profile_id):
```

Comment:
since `cls` is never used, maybe you want `@staticmethod` instead?

Reply:
Addressed with commit.